src/ethereum/cancun/fork.py

```diff
@@ -349,6 +349,7 @@ def validate_header(chain: BlockChain, header: Header) -> None:
         raise InvalidBlock
     if header.nonce != b"\x00\x00\x00\x00\x00\x00\x00\x00":
         raise InvalidBlock
+     # Post-merge forks do not allow ommers, so parent_header is always the last header in the chain.
```

Comment:
You need to simplify the logic at the beginning of the function (the one for determining the `parent_header`). Since there are no ommers, the only block whose header needs to be validated is the current block. Which implies that the `parent_header` is just the header of the last block of the chain.